test/pullrequest/e2e_test.go

```diff
@@ -0,0 +1,200 @@
+// Package pullrequest tests the pullrequest-init binary end-to-end by invoking
+// the binary and observing the resulting filesystem state.
+package pullrequest
```

Comment:
I intentionally wrote this in a way so that it could be executed as a unit test. :D 

The idea was that by default the proxy is spun up and all traffic stays local by default. Only if the proxy flag is disabled should this actually make external calls.

That said, I don't feel strongly about this so if we want to take the stance that if it shells out it should have an e2e label that sounds fine. Let me know what you think!